dockerHashing/src/main/scala/cromwell/docker/local/DockerCliClient.scala

```diff
@@ -0,0 +1,105 @@
+package cromwell.docker.local
+
+import scala.util.Try
+
+/**
+  * Wrapper around the docker cli.
+  * https://docs.docker.com/engine/reference/commandline/docker/
+  *
+  * An alternative to using REST calls directly to the docker engine that has several versions, for example:
+  * https://docs.docker.com/engine/api/v1.27/
+  */
+trait DockerCliClient {
+  /**
+    * Looks up a docker hash.
+    *
+    * @param dockerCliKey The docker hash to lookup.
+    * @return The hash if found, None if not found, and Failure if an error occurs.
+    */
+  def lookupHash(dockerCliKey: DockerCliKey): Try[Option[String]] = {
+    /*
+    The stdout contains the tab separated repository/tag/digest for __all__ local images.
+    Would be great to just get a single hash using the key... unfortunately
+    https://github.com/docker/docker/issues/29901
+     */
+    forRun("docker", "images", "--digests", "--format", """{{printf "%s\t%s\t%s" .Repository .Tag .Digest}}""") {
+      _.flatMap(parseHashLine).find(_.key == dockerCliKey).map(_.digest)
+    }
+  }
+
+  /**
+    * Pulls a docker image.
+    * @param dockerCliKey The docker hash to lookup.
+    * @return Failure if an error occurs.
+    */
+  def pull(dockerCliKey: DockerCliKey): Try[Unit] = {
+    forRun("docker", "pull", dockerCliKey.fullName) { _ => () }
```

Comment:
Hmm I don't know how to use `void` in this particular case. This is not a `map { _ => () }`. The `forRun` method takes a second curried argument that's a `Seq[String] => A`